server/routes/userRoutes.js

```diff
@@ -106,18 +93,56 @@ module.exports = (db) => {
     const { host_name } = req.body.params;
     dbHelper
       .getWebsiteIDByHostname(host_name)
-      // .getWebsiteIDByHostname("reddit.com")
-      .then((site) => dbHelper.addWebsiteToBlacklist(userId, site.id))
+      .then((site) => {
+        if (!site) {
+          // Removes '.com' from end of hostname for DB.
+          const remSuffix = host_name.split(".")[0];
+          // Capitalizes first letter of hostname
+          const name = remSuffix.charAt(0).toUpperCase() + remSuffix.slice(1);
+          const category = null;
+          dbHelper
+          .addWebsite(host_name, name, category)
+          .then((site) => {
+            dbHelper.addWebsiteToBlacklist(userId, site.id);
+          });
+        }
+        dbHelper.addWebsiteToBlacklist(userId, site.id);
+      })
       .catch((err) => res.json(err));
```

Comment:
you need to send back a 4xx status code, 400 is "bad request" so use that. Change all these to `res.status(400).json(err)`

Reply:
Fixed it on my end now